dot/parachain/availability-store/availability_store.go

```diff
@@ -456,32 +566,281 @@ func (av *AvailabilityStoreSubsystem) processMessages() {
 				}
 
 			case parachaintypes.ActiveLeavesUpdateSignal:
+				av.currentMessage = msg
 				av.ProcessActiveLeavesUpdateSignal()
 
 			case parachaintypes.BlockFinalizedSignal:
+				av.currentMessage = msg
 				av.ProcessBlockFinalizedSignal()
 
 			default:
-				logger.Error(parachaintypes.ErrUnknownOverseerMessage.Error())
+				// TODO(ed): handle unknown message type (if necessary)
+				if msg != nil {
+					logger.Infof("unknown message type %T", msg)
+					logger.Error(parachaintypes.ErrUnknownOverseerMessage.Error())
+				}
 			}
-
 		case <-av.ctx.Done():
 			if err := av.ctx.Err(); err != nil {
-				logger.Errorf("ctx error: %v\n", err)
+				logger.Errorf("ctx error: %v", err)
 			}
 			av.wg.Done()
 			return
+		case <-time.After(av.pruningConfig.PruningInterval):
+			av.pruneAll()
 		}
-
 	}
 }
 
 func (av *AvailabilityStoreSubsystem) ProcessActiveLeavesUpdateSignal() {
-	// TODO: #3630
+	now := av.clock.Now()
+	logger.Infof("ProcessActiveLeavesUpdateSignal %s", av.currentMessage)
+	activeLeave := av.currentMessage.(parachaintypes.ActiveLeavesUpdateSignal)
+
+	respChan := make(chan any)
+	message := util.ChainAPIMessage[util.BlockHeader]{
+		Message: util.BlockHeader{
+			Hash: activeLeave.Activated.Hash,
+		},
+		ResponseChannel: respChan,
+	}
+	response, err := util.Call(av.SubSystemToOverseer, message, message.ResponseChannel)
+	if err != nil {
+		logger.Errorf("sending message to get block header: %w", err)
```

Comment:
is there a reason why `ProcessActiveLeavesUpdateSignal` doesn't return an error? I don't think we should get in the practice of just logging errors that should actually be handled.

Reply:
Updated this to return error.